libecl/src/ecl_type.c

```diff
@@ -0,0 +1,204 @@
+/*
+   Copyright (C) 2017  Statoil ASA, Norway.
+
+   The file 'ecl_type.c' is part of ERT - Ensemble based Reservoir Tool.
+
+   ERT is free software: you can redistribute it and/or modify
+   it under the terms of the GNU General Public License as published by
+   the Free Software Foundation, either version 3 of the License, or
+   (at your option) any later version.
+
+   ERT is distributed in the hope that it will be useful, but WITHOUT ANY
+   WARRANTY; without even the implied warranty of MERCHANTABILITY or
+   FITNESS FOR A PARTICULAR PURPOSE.
+
+   See the GNU General Public License at <http://www.gnu.org/licenses/gpl.html>
+   for more details.
+*/
+
+#include <stdlib.h>
+#include <string.h>
+
+#include <ert/util/util.h>
+#include <ert/ecl/ecl_util.h>
+#include <ert/ecl/ecl_type.h>
+
+/*****************************************************************/
+/* The string names for the different ECLIPSE low-level
+   types.
+*/
+#define ECL_TYPE_NAME_CHAR     "CHAR"
+#define ECL_TYPE_NAME_C010     "C010"
+#define ECL_TYPE_NAME_FLOAT    "REAL"
+#define ECL_TYPE_NAME_INT      "INTE"
+#define ECL_TYPE_NAME_DOUBLE   "DOUB"
+#define ECL_TYPE_NAME_BOOL     "LOGI"
+#define ECL_TYPE_NAME_MESSAGE  "MESS"
+
+ecl_data_type * ecl_type_alloc_copy(const ecl_data_type * src_type) {
+    ecl_data_type * ecl_type;
+    ecl_type = util_malloc(ecl_type_get_sizeof_ctype(src_type));
+    memcpy(ecl_type, src_type, ecl_type_get_sizeof_ctype(src_type));
+    return ecl_type;
+}
+
+void ecl_type_free(ecl_data_type * ecl_type) {
+    if(ecl_type == NULL)
+        return;
+
+    free(ecl_type);
+}
+
+ecl_data_type ecl_type_get_data_type(const ecl_type_enum type, const size_t element_size) {
```

Comment:
This is function creates a *new* object - i.e. it is *not* a *get* function. I would suggest `ecl_type / ecl_type_make / ecl_type_create / .... `